### PR TITLE
feat: add rest api manager

### PR DESCRIPTION
**Pull request by andrewleap-optimizely**
## Summary
Provide an internal service for sending ODP events via REST api. 

- When dispatch is completed successfully, no action is required.
- Returns should_retry bool to allow retries on network/server errors

## Test plan
- added zaius_rest_api_manager_spec.rb

## Ticket
[OASIS-8450](https://optimizely.atlassian.net/browse/OASIS-8450)

### CODE REVIEW

File: lib/optimizely/odp/zaius_rest_api_manager.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+      begin
+        response = Helpers::HttpUtils.make_request(
+          url, :post, events.to_json, headers, Optimizely::Helpers::Constants::ODP_REST_API_CONFIG[:REQUEST_TIMEOUT], @proxy_config
+        )
+      rescue SocketError, Timeout::Error, Net::ProtocolError, Errno::ECONNRESET
+        log_failure('network error')
+        should_retry = true
+        return should_retry
+      rescue Errno::EINVAL, Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError => e
```
Comment (jaeopt) [inline lib/optimizely/odp/zaius_rest_api_manager.rb:50]: Wondering if this catches all other exceptions.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good! A couple of clarifications.

File: lib/optimizely/odp/zaius_rest_api_manager.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      should_retry = false
+      url = "#{api_host}/v3/events"
+
+      headers = {'Content-Type' => 'application/json', 'x-api-key' => api_key.to_s}
+
+      begin
+        response = Helpers::HttpUtils.make_request(
+          url, :post, events.to_json, headers, Optimizely::Helpers::Constants::ODP_REST_API_CONFIG[:REQUEST_TIMEOUT], @proxy_config
+        )
+      rescue SocketError, Timeout::Error, Net::ProtocolError, Errno::ECONNRESET
```
[the remark this answers is not in the document]
Comment (andrewleap-optimizely) [reply lib/optimizely/odp/zaius_rest_api_manager.rb]: Yea agreed. Updated with seemingly relevant exceptions (documentation is not great).

File: lib/optimizely/odp/zaius_rest_api_manager.rb (hunk 1 above)
Comment (andrewleap-optimizely) [reply lib/optimizely/odp/zaius_rest_api_manager.rb]: Changed to ruby base exception

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by zashraf1985 — APPROVED** [review]
Looks Great!

**Pull request merged**